packages/core/src/dialects/mssql/query-generator-typescript.ts

```diff
@@ -83,4 +83,36 @@ export class MsSqlQueryGeneratorTypeScript extends AbstractQueryGenerator {
       this.quoteTable(tableName),
     ]);
   }
+
+  getForeignKeyQuery(tableName: TableNameOrModel, columnName?: string) {
+    const table = this.extractTableDetails(tableName);
+
+    // TODO: get the database from the provided tableName (see #12449)
+    const catalogName = this.sequelize.config.database;
+    const escapedCatalogName = this.escape(catalogName);
+
+    return joinSQLFragments([
+      'SELECT constraint_name = OBJ.NAME,',
+      'constraintName = OBJ.NAME,',
```

Comment:
It's selecting `OBJ.NAME` as both `constraint_name` and `constraintName`. Other dialects use `constraintName`, I propose to remove `constraint_name`